src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -562,17 +574,24 @@ export async function refundRequestIsValid(
     return { valid: false, reason: "Unable to find matching deposit" };
   }
 
-  // Verify that the refundToken maps to a known HubPool token.
-  // Note: the refundToken must be valid at the time of the Fill *and* the RefundRequest.
-  // @todo: Resolve to the HubPool block number at the time of the RefundRequest ?
-  const hubPoolBlockNumber = hubPoolClient.latestBlockNumber ?? hubPoolClient.deploymentBlock - 1;
+  // Verify that the refundToken maps to a known HubPool token and is the correct
+  // token for the chain where the refund was sent from.
+  // Note: the refundToken must be valid at the time the deposit was sent.
   try {
-    hubPoolClient.getL1TokenCounterpartAtBlock(repaymentChainId, refundToken, hubPoolBlockNumber);
+    const l1TokenForFill = hubPoolClient.getL1TokenCounterpartAtBlock(
+      fill.destinationChainId,
+      fill.destinationToken,
+      deposit.quoteBlockNumber
+    );
+    const expectedRefundToken = hubPoolClient.getDestinationTokenForL1Token(l1TokenForFill, repaymentChainId);
+    if (expectedRefundToken !== refundToken) {
+      return { valid: false, reason: `Refund token does not map to expected refund token ${refundToken}` };
+    }
   } catch {
-    return { valid: false, reason: `Refund token unknown at HubPool block ${hubPoolBlockNumber}` };
+    return { valid: false, reason: `Refund token unknown at deposit quote block ${deposit.quoteBlockNumber}` };
```

Comment:
```suggestion
    return { valid: false, reason: `Refund token unknown at HubPool block ${deposit.quoteBlockNumber}` };
```

Thoughts on the message as is? `HubPool block` is more obvious than `deposit quote block` IMO.